### ISSUE TITLE
Data failed to compile errors

### ISSUE THREAD

**Issue by SteveZman** [saltstack/salt#48056]
### Description of Issue/Question
Working to set up a salt environment and I repeatedly encounter this error when building state files.  Unable to determine what is causing the error
### Setup
using a basic .sls file for testing:
install_apache:
  pkg.installed:
    - name: apache2
command issued and results:
lzimmerm@ntsinfsalt101 salt]$ sudo salt -vv '*lsus*' state.sls apache
[sudo] password for lzimmerm: 
Executing job with jid 20180611111533081852
-------------------------------------------

ntsinflsus102:
    Data failed to compile:
----------
    The function "state.sls" is running as PID 19696 and was started at 2018, Jun 11 11:15:33.081852 with jid 20180611111533081852
ERROR: Minions returned with non-zero exit code

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
immerm@ntsinfsalt101 salt]$ sudo salt --versions-report
[sudo] password for lzimmerm: 
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Feb 20 2018, 09:19:12)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: redhat 7.5 Maipo
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-862.2.3.el7.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 7.5 Maipo

**Comment by gtmanfred** [comment]
So, the error here is that you already have a state running on that other minion.

You can only run one state run on a minion at a time.

You can see what is running on the salt minion using

`salt \* saltutil.is_running`

And Kill any running jobs with 

`salt \* saltutil.kill_job <jid>`

**Comment by SteveZman** [comment]
Checked for running states – same results:

[lzimmerm@ntsinfsalt101 ~]$ sudo salt '*lsus*' saltutil.is_running *
[sudo] password for lzimmerm:
ntsinflsus102:
ntsinflsus102:
[lzimmerm@ntsinfsalt101 ~]$ sudo salt '*lsus*' saltutil.is_running state
[sudo] password for lzimmerm:
ntsinflsus102:
ntsinflsus102:
[lzimmerm@ntsinfsalt101 ~]$ sudo salt '*lsus*' state.sls apache
[sudo] password for lzimmerm:
ntsinflsus102:
    Data failed to compile:
----------
    The function "state.sls" is running as PID 27879 and was started at 2018, Jun 12 09:47:12.971485 with jid 20180612094712971485
ERROR: Minions returned with non-zero exit code
[lzimmerm@ntsinfsalt101 ~]$

One note – can there be an issue due to a duplicate name?  We have one server named ntsinflsus102 yet two seem to appear here. Could that be an issue?

Thanks
Steve

**Comment by gtmanfred** [comment]
That could do it, it looks like you might have two minion processes running on the same minion, with the same name.

Login to that minion, and stop the salt-minion service, then see if any salt-minion processes are still running, and that could cause the issue.

**Comment by SteveZman** [comment]
Seem to be getting a similar error – it appears that there is only one minion service running:

root@ntsinflsus102 ~]# systemctl status salt-minion
● salt-minion.service - The Salt Minion
   Loaded: loaded (/usr/lib/systemd/system/salt-minion.service; disabled; vendor preset: disabled)
   Active: active (running) since Tue 2018-06-12 10:47:45 CDT; 1min 52s ago
     Docs: man:salt-minion(1)
           file:///usr/share/doc/salt/html/contents.html
           https://docs.saltstack.com/en/latest/contents.html
Main PID: 13540 (salt-minion)
   CGroup: /system.slice/salt-minion.service
           ├─13540 /usr/bin/python /usr/bin/salt-minion
           ├─13544 /usr/bin/python /usr/bin/salt-minion
           └─13548 /usr/bin/python /usr/bin/salt-minion

Jun 12 10:47:45 ntsinflsus102.ntspay.local systemd[1]: Starting The Salt Minion...
Jun 12 10:47:45 ntsinflsus102.ntspay.local systemd[1]: Started The Salt Minion.

But still getting a similar error:

[lzimmerm@ntsinfsalt101 salt]$ sudo salt '*lsus*' state.apply miniontest
[sudo] password for lzimmerm:
Sorry, try again.
[sudo] password for lzimmerm:
ntsinflsus102:
    Data failed to compile:
----------
    The function "state.apply" is running as PID 13837 and was started at 2018, Jun 12 10:49:28.152286 with jid 20180612104928152286
ERROR: Minions returned with non-zero exit code
[lzimmerm@ntsinfsalt101 salt]$

Is there a line of code I am not putting into these simple state files that is needed to let failed requests exit gracefully?

Make sure the mysql service is running:
  service.running:
    - name: salt-minion
~

Appreciate the support.
Steve

**Comment by gtmanfred** [comment]
Can you check the actual process list with `ps aux| grep salt-minion`? and make sure there aren't extra minion jobs running besides the one from systemctl status salt-minion?

**Comment by SteveZman** [comment]
Here we go – not sure how it should look.

[root@ntsinflsus102 ~]# systemctl status salt-minion
● salt-minion.service - The Salt Minion
   Loaded: loaded (/usr/lib/systemd/system/salt-minion.service; disabled; vendor preset: disabled)
   Active: active (running) since Tue 2018-06-12 10:47:45 CDT; 1min 52s ago
     Docs: man:salt-minion(1)
           file:///usr/share/doc/salt/html/contents.html
           https://docs.saltstack.com/en/latest/contents.html
Main PID: 13540 (salt-minion)
   CGroup: /system.slice/salt-minion.service
           ├─13540 /usr/bin/python /usr/bin/salt-minion
           ├─13544 /usr/bin/python /usr/bin/salt-minion
           └─13548 /usr/bin/python /usr/bin/salt-minion

Jun 12 10:47:45 ntsinflsus102.ntspay.local systemd[1]: Starting The Salt Minion...
Jun 12 10:47:45 ntsinflsus102.ntspay.local systemd[1]: Started The Salt Minion.
[root@ntsinflsus102 ~]# ps aux | grep salt-minion
root     13540  0.0  0.1 313612 21328 ?        Ss   10:47   0:00 /usr/bin/python /usr/bin/salt-minion
root     13544  0.1  0.2 716384 48196 ?        Sl   10:47   0:02 /usr/bin/python /usr/bin/salt-minion
root     13548  0.0  0.1 413840 22396 ?        S    10:47   0:00 /usr/bin/python /usr/bin/salt-minion
root     20678  0.0  0.0 112708   984 pts/1    S+   11:12   0:00 grep --color=auto salt-minion
root     22309  0.0  0.2 792024 47296 ?        Sl   May16  29:46 /usr/bin/python /bin/salt-minion -d
[root@ntsinflsus102 ~]#

**Comment by gtmanfred** [comment]
Yup, you have an extra minion that was started with `salt-minion -d`
```
root 22309 0.0 0.2 792024 47296 ? Sl May16 29:46 /usr/bin/python /bin/salt-minion -d\
```

do a `kill -9 22309` and that should fix your problem.

**Comment by SteveZman** [comment]
That did it!  Thanks so much.

Steve
